app/assets/javascripts/discourse/app/controllers/preferences/sidebar.js

```diff
@@ -20,12 +33,21 @@ export default class extends Controller {
 
     this.model.set("sidebar_tag_names", this.selectedSidebarTagNames);
 
+    this.model.set(
+      "user_option.sidebar_topic_destination",
+      this.newSidebarTopicDestination
+    );
+
     this.model
       .save()
       .then((result) => {
         if (result.user.sidebar_tags) {
           this.model.set("sidebar_tags", result.user.sidebar_tags);
         }
+        this.model.set(
+          "user_option.sidebar_topic_destination",
+          result.user.user_option.sidebar_topic_destination
```

Comment:
I'm not sure if this is necessary since we've already set it above and I don't think we expect the server to set the value to something different.